### DISCUSSION TITLE
dodanie-cloudflare-tunel-router-openwrt/

### DISCUSSION THREAD

**Discussion by giscus** [idarek/dariusz.wieckiewicz.org#111] [General]
# dodanie-cloudflare-tunel-router-openwrt/

W tym wpisie opisuję jak utworzyć tunel Cloudflare dla routera z OpenWrt, aby móc połączyć się z naszą siecią lokalną, bez konieczności VPN, bez otwierania portów oraz za CG-NAT.

https://dariusz.wieckiewicz.org/dodanie-cloudflare-tunel-router-openwrt/

**Comment by karol27** [comment]
Zrobiłem według poradnika i na końcu jak chciałem zalogować się do luci do miałem Bad Gateway. Po uruchomieniu tunelu fragmenty konsoli świadczyły o tym że coś jeszcze brakuje:
WRN "No ingress rules were defined in provided config (if any) nor from the cli, cloudflared will return 503 for all incoming HTTP requests"
WRN "The user running cloudflared process has a GID (group ID) that is not within ping_group_range. You might need to add that user to a group within that range, or instead update the range to encompass a group the user is already in by modifying /proc/sys/net/ipv4/ping_group_range. Otherwise cloudflared will not be able to ping this network error="Group ID 0 is not between ping group 1 to 0"
Nie bardzo wiem co z tym zrobić i co tam trzeba pododawać. Ktoś podpowie?

**Comment by idarek** [comment]
Bad Gateway świadczy o nieustawionym dostępie do sieci lokalnej, przejrzyj jeszcze raz:
- https://dariusz.wieckiewicz.org/dodanie-cloudflare-tunel-router-openwrt/#dodanie-adresów-naszej-sieci-lokalnej

Jeżeli chodzi o Ping to również, zobacz:
- https://dariusz.wieckiewicz.org/dodanie-cloudflare-tunel-router-openwrt/#dodanie-obsługi-pakietów-utp-i-icmp-ping

Podejrzewam że masz gdzieś problem w ustawieniach po stronie CloudFlare. Zalecam przejście przez cały poradnik jeszcze raz i sprawdzić czy wszystko jest wprowadzone tak jak opisane.

**Comment by mgmorpheus77** [comment]
Witam
Mam problem już na samym początku tego poradnika. Nie mogę wstawić a raczej zapisać tej ścieżki w pliku

net.core.rmem_max=2500000
net.ipv4.ping_group_range=0 2147483647

gdzie robię błąd? Ktoś pomoże?

**Reply by idarek** [reply]
Jak wydajesz komendę `touch /etc/sysctl.d/99-fix-buffer-and-ping.conf` czy masz jakiś błąd?

**Comment by mgmorpheus77** [comment]
Nie, nie ma błędu. Tworzy ten plik bo później mogę do niego wejść vim, ale jeśli cokolwiek dodam do niego to komenda :wq nie działa, ale jeśli w ogóle nic nie będę w nim robił komenda :wq działa bez zarzutu

Sent with Spark

**Reply by idarek** [reply]
`vim /etc/sysctl.d/99-fix-buffer-and-ping.conf`

Naciskasz na klawiaturze literę <kbd>i</kbd>, wówczas VIM przechodzi w tryb interaktywny.
Dodajesz obie linijki, naciskasz <kbd>Esc</kbd> aby wyjść z trybu interaktywnego, wówczas zapis i wyjście komendą `:wq`.

**Comment by mgmorpheus77** [comment]
Kolego jesteś wielki, mam wielki szacunek dla takich mózgów z IT
Ja z doskoku czasami jak mnie najdzie to grzebie sobie w komputerach zwykle jak mam za dużo wolnego czasu i potrzebuje się trochę zdenerwować 🙂
Jeśli masz gdzieś na Swoim blogu opcję dotacji to chętnie wspomogę
Jeśli będę miał w następnych krokach jakiś problem to pozwolę sobie napisać
I przepraszam za zawracanie gitary w prostych sprawach
Pozdrawiam i raz jeszcze dziękuję

Sent with Spark

**Comment by mgmorpheus77** [comment]
To znowu ja :) szybko
Teraz mam problem z aplikacją WARP
Nie mogę się zalogować Zero Trust
Pokazuje się pusta strona

Wszystko do tego miejsca wykonane z Twojego poradnika bez błędów

Sent with Spark

**Reply by idarek** [reply]
A czy w `Cloudflare Zero Trus > Access > Tunnels` tunel wyświetla się jako działający?
Podczas połączenia aplikacji Zero Trust wpisałeś nazwę swojej drużyny poprawnie?

Jeżeli nie pamiętasz, możesz ją znaleźć (i zmienić) w `Settings > Custom pages`

**Comment by mgmorpheus77** [comment]
Tak, jest na zielono
Nazwa drużyny prawidłowo wpisana
Nie wiem, czy to ma znaczenie, ale wklejając ten adres z WARPa do zwykłej przeglądarki wyskakuje error 404 strona nie istnieje
Coś czuję, że gdzieś jakiś błąd jest na stronie Cloudflare
I tak jestem mega zadowolony, że do tego etapu doszedłem
Jeśli masz jakieś sugestie co sprawdzić to będę wdzięczny

Sent with Spark

**Comment by mgmorpheus77** [comment]
Tak mnie teraz oświeciło, czy w podaniu adresacji nie popełniłem błędu
Ma znaczenie port z którym się łącze z routerem? 22 a w adresacji wpisałem 24 z rozpędu kopiując Twoje wpisy

Sent with Spark

**Reply by idarek** [reply]
Jaki port?
Masz na myśli /24 maskę podsieci. Musisz ustawić tak, aby pasowało do twojej sieci.

Posprawdzaj wszystko jeszcze raz.

Aby sprawdzić czy wszystko działa, wpisz w przeglądarce `https://XXX.cloudflareaccess.com`

gdzie XXX to nazwa twoje drużyny.

Powinieneś zobaczyć stronę z następującym tekstem.

```
Welcome!
Please contact your administrator to enable the Access App Launch on your account
```

**Comment by mgmorpheus77** [comment]
Tak maska, mój błąd
Mam u siebie 22 ale wg Twojego poradnika usunąłem te z 24 i nie wiem czy to ma znaczenie
Muszę chyba tutaj w tym miejscu raz jeszcze wykonać to krok po kroku
Dzięki za poświęcony czas, dam na pewno znać, czy się udało
Dziękuję

Sent with Spark

**Comment by mgmorpheus77** [comment]
Pomyliłem pojęcia x_x totalnie
Sorry za zamieszanie
Muszę odświeżyć wiedzę :/
Posprawdzam wszystko na stronie Cloudflare i się odezwę, dzięki

Sent with Spark

**Comment by mgmorpheus77** [comment]
To chyba w moim przypadku jest grubszy problem, aż tak się nie znam, więc opiszę dokładniej
Dostawca dał mi mostek i od tego mostka mam dopiero swój router
Na swoim routerze mam adresację 192.168.100.0/24 (maskę już sprawdziłem :D) a z mostka jeśli dobrze rozumuję jest 91.236.xx.xx
Czy mam dodać jeszcze w Cloudflare tą adresację z mostka? Wybacz Darek jeśli moje pytania wydają Ci się śmieszne, ale jestem totalnym amatorem próbującym coś zdziałać ;)

Sent with Spark

**Reply by idarek** [reply]
Cloudflare to Tunel, który omija wszelkich pośredników, więc nie interesuje cię co się dzieje za twoim routerem. Ustaw wszystko zgodnie z moim poradnikiem.

W części **Private Network** dodajesz `192.168.100.0/24`.

Dalej w **Split Tunnels**, jeżeli wybierzesz opcję **Exclude IPs and domains** to wyrzucasz tylko `192.168.0.0/16` i `192.0.0.0/24`

Jeżeli za bardzo namieszałeś w tej sekcji to przyciśnij przycisk na górze **Restore default entries** i zacznij od nowa.

I całość powinna działać, powinieneś sobie poradzić.

> Jeżeli chcesz (ale najpierw spróbuj sam) to mogę ci pomóc sprawdzić ustawienia po stronie CloudFlare. Możesz mnie dodać w nim jako dodatkowy administrator, ale so to tego to możemy wymienić się na mailu.

**Comment by mgmorpheus77** [comment]
Muszę to zrobić sam, przynajmniej się postaram
Sam sflashowałem router Xiaomi pod OpenWRT, gdzie już myślałem, że to mój Everest możliwości :) ale dałem rade
Co do pośredników to właśnie sięgnąłem po Twoj drugi poradnik, bo wcześniej chciałem użyć Wireguarda, ale okazało się, że porty usługodawcy są blokowane i trzeba je przekierować, więc odpuściłem temat na kilka miesięcy :)
Teraz znalazłem Twoj kolejny poradnik, który pomija te porty, NATa itd
Odezwę się po weekendzie, bo dziś na imprezę się udaję, więc raczej tego nie ogarnę
Dziękuję bardzo za poświęcony czas jak na razie

Sent with Spark

**Comment by mgmorpheus77** [comment]
Cześć
Chyba jednak sobie sam z tym nie poradzę, ale ostatnie dane które mi przyszły na myśl
Na koncie CF mam też tunel dla Homeassistant niezmigrowany, czyon może być problemem
Może nowe konto na CF będzie rozwiązaniem?
Pozdrawiam

Sent with Spark

**Reply by idarek** [reply]
Posiadanie innego tunelu nie powinno powodować problemu. Wyślę ci maila z informacjami jak dodać mnie do twojego CloudFlare aby mógł zobaczyć czy wszystko działa. Pozdrawiam

**Comment by mgmorpheus77** [comment]
Witam
To znowu ja niestety :/ i mam nadzieję, że i tym razem będziesz pomocny Darku. 
Ehhh, nic nie grzebałem, tzn. zmieniłem hasło wifi, ale i tak później wróciłem z powrotem do tego co było i się zaczęło ;/
Rozłącza mi zdalny tunel zaraz po wyjściu z konsoli putty, gdzie szukać przyczyny?
Uruchamiam tunel z powrotem, ale zaraz po wyjściu z konsoli się rozłącza, nie wiem totalnie co może być przyczyną

**Reply by mgmorpheus77** [reply]
Sorry za zamieszanie, ale wydaje się chyba, że zaradziłem problemowi x_x złą komendą uruchamiałem tunel. Na razie wszystko jest ok :D
